include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -184,1400 +170,959 @@ class alignment_file_input_format_REMOVEME<format_sam>
+    alignment_file_header<> default_header{};
+
+    //!\brief Tracks whether reference information (\@SR tag) were found in the SAM header
+    bool ref_info_present_in_header{false};
+
+    //!brief Returns a reference to dummy if passed a std::ignore.
+    std::string_view const & default_or(detail::ignore_t) const noexcept
     {
-        static_assert(detail::decays_to_ignore_v<ref_offset_type> ||
-                      detail::is_type_specialisation_of_v<ref_offset_type, std::optional>,
-                      "The ref_offset must be a specialisation of std::optional.");
-
-        auto stream_view = views::istreambuf(stream);
-        auto field_view = stream_view | views::take_until_or_throw_and_consume(is_char<'\t'>);
-
-        // these variables need to be stored to compute the ALIGNMENT
-        int32_t ref_offset_tmp{};
-        value_type_t<decltype(header.ref_ids())> ref_id_tmp{};
-        [[maybe_unused]] int32_t offset_tmp{};
-        [[maybe_unused]] int32_t soft_clipping_end{};
-        [[maybe_unused]] std::vector<cigar> tmp_cigar_vector{};
-        [[maybe_unused]] int32_t ref_length{0}, seq_length{0}; // length of aligned part for ref and query
-
-        // Header
-        // -------------------------------------------------------------------------------------------------------------
-        if (is_char<'@'>(*std::ranges::begin(stream_view))) // we always read the header if present
-        {
-            read_header(stream_view, header, ref_seqs);
+        return dummy;
+    }
 
-            if (std::ranges::begin(stream_view) == std::ranges::end(stream_view)) // file has no records
-                return;
-        }
+    //!brief Returns the input unchanged.
+    template <typename t>
+    decltype(auto) default_or(t && v) const noexcept
+    {
+        return std::forward<t>(v);
+    }
 
-        // Fields 1-5: ID FLAG REF_ID REF_OFFSET MAPQ
-        // -------------------------------------------------------------------------------------------------------------
-        read_field(field_view, id);
+    using format_sam_base::read_field; // inherit read_field functions from format_base explicitly
 
-        read_field(field_view, flag);
+    template <typename stream_view_type, typename value_type>
+    void read_sam_dict_vector(seqan3::detail::sam_tag_variant & variant,
+                              stream_view_type && stream_view,
+                              value_type value);
 
-        read_field(field_view, ref_id_tmp);
-        check_and_assign_ref_id(ref_id, ref_id_tmp, header, ref_seqs);
+    template <typename stream_view_type>
+    void read_field(stream_view_type && stream_view, sam_tag_dictionary & target);
 
-        read_field(field_view, ref_offset_tmp);
-        --ref_offset_tmp; // SAM format is 1-based but SeqAn operates 0-based
+    template <typename stream_it_t, typename field_type>
+    //!\cond
+        requires std::ranges::forward_range<field_type>
```

Comment:
use intermediate concept syntax

Reply:
you mean the short syntax within the template syntax?

Reply:
Don't use "short sytax" for this. Short syntax is `std::ranges::forward_range auto var_name`.